tests/kernel/pipe/pipe/src/test_pipe.c

```diff
@@ -874,3 +883,41 @@ void test_pipe_put_min_xfer(void)
 	zassert_equal(res, 0, "did not write min_xfer");
 	zassert_true(bytes_written >= 1, "did not write min_xfer");
 }
+
+/**
+ * @brief Test defining and initializing pipes at run time
+ * @details Check if the kernel provided a mechanism for defining and
+ * initializing pipes at run time.
+ * - Define and initialize a pipe at run time
+ * - Using this pipe to transfer data and verify it
+ * @ingroup kernel_pipe_tests
+ * @see k_pipe_init()
+ */
+void test_pipe_define_at_runtime(void)
+{
+	unsigned char __aligned(4) ring_buffer[PIPE_SIZE];
+	struct k_pipe pipe;
+	size_t written, read;
+
+	/*Define and initialize pipe at run time*/
+	k_pipe_init(&pipe, ring_buffer, sizeof(ring_buffer));
+
+	/* initialize the tx buffer */
+	for (int i = 0; i < sizeof(tx_buffer); i++) {
+		tx_buffer[i] = i;
+	}
+
+	/*Using test_pipe which define and initialize at run time*/
+	zassert_equal(k_pipe_put(&pipe, &tx_buffer,
+			PIPE_SIZE, &written,
+			PIPE_SIZE, K_NO_WAIT), RETURN_SUCCESS, NULL);
+
+	zassert_equal(k_pipe_get(&pipe, &rx_buffer,
+			PIPE_SIZE, &read,
+			PIPE_SIZE, K_NO_WAIT), RETURN_SUCCESS, NULL);
+
+	zassert_true(rx_buffer_check(rx_buffer, read) == read,
+			"Bytes read are not match.");
+
+	ztest_test_pass();
```

Comment:
Remove this line.  Returning from a top-level ztest function without having flagged a failure is a "pass" result, by definition.

Reply:
@andyross ,  Have removed the pass line,  thank you for reviewing again.